### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by Tafloninno**
### In this PR i completed the following

Splash screen

-  A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

   
Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).

- The user can log into the app using email and password.

- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

Home page (categories page)

- When the user logs in, they are presented with the categories page.

- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.

- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.

- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).

- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).

- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.

- When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new transaction" page
> The user fills out a form to create a new transaction with:

- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)

> The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.

> When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

Testing requirements

- Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,129 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-* Database initialization
+## 🚀 Live Demo <a name="live-demo" ></a>
 
-* How to run the test suite
+- [Live Demo](https://buddy-4izw.onrender.com)
 
-* Services (job queues, cache servers, search engines, etc.)
+- [Presentation video](https://www.loom.com/share/2ed6d21688d54f91b1878f3e13a539b3)
```
Comment (zieeco) [inline README.md]: - Hey, you've done a great job, just a few things to improve and your project would be approved. Please, kindly remember to show a piece of code you are proud of to meet this part of the video requirement. kindly record another video and add this missing part of the requirement. :pray: 

image

File: Gemfile [hunk 2]
```diff
@@ -8,6 +8,8 @@ gem 'rails', '~> 7.0.5'
 
 gem 'rubocop', '>= 1.0', '< 2.0'
 
+gem 'devise'
+
```
Comment (zieeco) [inline Gemfile:12]: - Hey, please, let's not forget to add some authorization rules by adding the CanCanCan gem, to add the `ability.rb` file for a user to meet the project requirement below:

image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Tafloninno,




Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Descriptive PR. ✔️
- Good commit messages :+1: 
- Nice design :+1: 

## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OyePriscilla — APPROVED** [review]
Hi @Tafloninno 👍, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Used `cancancan gem` for authorization ✔️ 
Video presentation engaging ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**